lib/view_component/global_output_buffer.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module GlobalOutputBuffer
+    def render_in(view_context, &block)
+      unless view_context.output_buffer.is_a?(OutputBufferStack)
+        # use instance_variable_set here to avoid triggering the code in the #output_buffer= method below
+        view_context.instance_variable_set(:@output_buffer, OutputBufferStack.new(view_context.output_buffer))
+      end
+
+      @output_buffer = view_context.output_buffer
+
+      super
+    end
+
+    def perform_render
+      # HAML unhelpfully assigns to @output_buffer directly, so we hold onto a reference to
+      # it and restore @output_buffer when the HAML engine is finished. In non-HAML cases,
+      # @output_buffer and orig_buf will point to the same object, making the reassignment
+      # statements no-ops.
+      orig_buf = @output_buffer
+      @output_buffer.push
+      result = render_template_for(@__vc_variant).to_s + _output_postamble
+      @output_buffer = orig_buf
+      @output_buffer.pop
+      result
+    end
+
+    def output_buffer=(other_buffer)
+      if @output_buffer && other_buffer
+        @output_buffer.replace(other_buffer)
+      else
+        # :nocov:
+        @output_buffer = other_buffer
+        # :nocov:
+      end
+    end
+
+    def with_output_buffer(buf = nil)
```

Comment:
Do we need to override this?

If-so, is there any issue with partials using the existing `with_output_buffer` method with the output buffer stack?

Reply:
Yeah, because any view that calls `#capture` will call this method. The original temporarily swaps `self.output_buffer` whereas the buffer stack needs to be `#push`ed to. As far as I can tell, form helpers and the fragment caching stuff are the only things that `#capture` though. In other words if you're not doing either of those things, this code never gets invoked.